From 210e1ccb98535cdafaaca10736cfd894beaa3c3f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 30 Sep 2021 20:47:19 +0300
Subject: [PATCH] [SW-57547] gaudi2: arc: handle error in preboot copy

Currently if preboot dma from SPI to SRAM or verification of preboot
binary is failed, valid loadaddr is sent along with success as part
of ppboot_load_preboot_img function. This patch add support to return
false if preboot image dma or verification is failed which will initiate
recovery flow instead of starting preboot image.

Change-Id: I637021b53fd00994f817cf413978f83b97ca8c6c
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index e0b0e1f..000c91d 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -237,6 +237,7 @@
 		ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_fuse);
 		if (!ret && config_fuse != EFUSE_SYSCFG_DISABLE_TWI_IF_MASK)
 			security_i2c_s_enable();
+		return 0;
 	}
 
 	return PREBOOT_TEXT_BASE;
@@ -265,6 +266,10 @@
 	set_spi_dma_br(SPI_DMA_BAUDR_SRAM_DMA);
 	*loadaddr = copy_preboot();
 	set_spi_dma_br(SPI_DMA_BAUDR_DEFAULT);
+
+	if (!*loadaddr)
+		return false;
+
 	return true;
 }
 
